src/main/kotlin/com/bantolomeus/service/BookService.kt

```diff
@@ -16,84 +16,87 @@ class BookService(private val bookRepository: BookRepository,
 
     fun createBook(bookDTO: BookDTO): BookDTO {
         val savedBookDTO = bookRepository.saveBookIfItNotExists(bookDTO)
-        if (savedBookDTO.currentPage > 0L) {
+        if (savedBookDTO.currentPage > 0) {
             saveBookUpdate(savedBookDTO)
         }
         return bookDTO
     }
 
     fun updateBook(bookUpdate: BookUpdateInputDTO, bookName: String): BookUpdatesFileDTO {
-        var response = BookUpdatesFileDTO()
-        var books = bookRepository.getBooks().books.filter { it.name != bookName }
-        var oldPage = Long.MAX_VALUE
-        val foundBook = bookRepository.getBookByName(bookName)
+        val foundBook = bookRepository.getBookByName(bookName) ?: return BookUpdatesFileDTO()
 
-        if (foundBook.name == bookName) {
-            foundBook.let {
-                oldPage = it.currentPage
-                it.currentPage = bookUpdate.currentPage
-                if (it.currentPage >= it.pagesTotal) {
-                    it.readTime = (Date().time - dateFormat.parse(it.dateStarted).time) / DIVISOR_FOR_DAY
-                }
-            }
-            books = mutableListOf(foundBook) + books.toMutableList()
+        if (foundBook.currentPage == foundBook.pagesTotal) {
+            foundBook.readTime = ((Date().time - dateFormat.parse(foundBook.dateStarted).time) / DIVISOR_FOR_DAY) + 1
```

Comment:
I do not want to fix the wrong readTime calculation now, because then I have some books with the correct and some with the wrong read time. I want to migrate the wrong ones along the way with the change. I want a helper method to correct the readTime from already finished books, because they are all wrong and it is easier to fix this state than a mix of wrong and correct readTime data.
So please remove the ```+ 1``` here
I have already the issue https://github.com/Bantolomeus/bookmaster4000mkii/issues/97 for it. Lets fix it there please

Reply:
haha, you tricked yourself. I didn't change this. This PR is showing the whole diff to master. If you look only at my commit (https://github.com/Bantolomeus/bookmaster4000mkii/commit/1999537bedbb980c640254d4a8d0ef8939e5daab#diff-aa2ee0a8339742d80054ccc579011d90L35) you see that I found it already with the `+ 1` ;-) 

I will leave it like this

Reply:
I changed it myself